stdlib/public/core/Result.swift

```diff
@@ -0,0 +1,202 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// A value that represents either a success or failure, capturing associated
+/// values in both cases.
+@_frozen
+public enum Result<Value, Error> {
+  /// A success, storing a `Value`.
+  case success(Value)
+  
+  /// A failure, storing an `Error`.
+  case failure(Error)
+  
+  /// The stored value of a successful `Result`. `nil` if the `Result` was a
+  /// failure.
+  public var value: Value? {
+    switch self {
+    case let .success(value): return value
+    case .failure: return nil
+    }
+  }
+  
+  /// The stored value of a failure `Result`. `nil` if the `Result` was a
+  /// success.
+  public var error: Error? {
+    switch self {
+    case let .failure(error): return error
```

Comment:
Wrap to next line here and in the next line.
```suggestion
    case let .failure(error): 
      return error
```